### PR TITLE
First support for Bittrex exchange

### PR DESCRIPTION
**Pull request by hugues31**
I will add support for private requests soon (and add missing public methods)

### CODE REVIEW

File: src/bittrex/api.rs [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+        // TODO: implement correctly the TLS error in error_chain.
+        let ssl = match NativeTlsClient::new() {
+            Ok(res) => res,
+            Err(_) => return Err(ErrorKind::TlsError.into()),
+        };
+        let connector = HttpsConnector::new(ssl);
+
+        Ok(BittrexApi {
+               last_request: 0,
+               api_key: creds.get("api_key").unwrap_or_default(),
```
Comment (crackcomm) [inline src/bittrex/api.rs]: I am just thinking here - what about `?`, what about invalid keys?

File: src/bittrex/credentials.rs [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+            .get(name)
+            .ok_or_else(|| ErrorKind::MissingField(name.to_string()))?;
+        let api_key = helpers::get_json_string(json_obj, "api_key")?;
+        let api_secret = helpers::get_json_string(json_obj, "api_secret")?;
+        let exchange = {
+            let exchange_str = helpers::get_json_string(json_obj, "exchange")?;
+            Exchange::from_str(exchange_str)
+                .chain_err(|| ErrorKind::InvalidFieldValue("exchange".to_string()))?
+        };
+
```
Comment (crackcomm) [inline src/bittrex/credentials.rs]: Okay, I can see how this question should be addressed here.
What I am trying to say is I would rather take certain measures to make sure configuration was not misinterpreted and check for extra keys which are not valid configuration keys like `api_secet` etc.

File: src/bittrex/api.rs (hunk 1 above)
Comment (hugues31) [reply src/bittrex/api.rs]: That could be nice. I'll wait finishing Bittrex implementation and next I'll be looking at how to improve security and optimization

File: src/bittrex/credentials.rs (hunk 2 above)
Comment (hugues31) [reply src/bittrex/credentials.rs]: Thanks for your input, I'll think about it :)

File: src/bittrex/api.rs (hunk 1 above)
Comment (crackcomm) [reply src/bittrex/api.rs]: @hugues31 I would like to help you, maybe you could create issue for that?

Comment (hugues31) [reply src/bittrex/api.rs]: @crackcomm Sure!

**Comment by hugues31** [comment]
I merge this PR (travis build failed due to a network error)

**Pull request merged**